### PR TITLE
feat: try to use the S3 tiles version if  available instead of returning 503

### PR DESCRIPTION
**Pull request by juanmahidalgo**
# Atlas Server - S3 Redirect Enhancement

## Context
Currently, when the map is not ready, the server returns a 503 status code. However, we have a feature flag that allows redirecting to S3 where we store cached versions of the tiles. This PR enhances the logic to use S3 redirects when available, even when the map is not ready.

## Changes
- Modified the tiles handlers (`createTilesRequestHandler` and `createLegacyTilesRequestHandler`) to:
  1. Check for S3 redirect feature flag first
  2. If enabled and S3 URL is available, redirect to S3 (regardless of map ready state)
  3. If enabled but no S3 URL available, fall back to map state check
  4. If disabled, maintain original behavior

### Logic Flow
When map is not ready:
- S3 feature OFF → 503 "Not ready"
- S3 feature ON + S3 URL available → 301 redirect to S3
- S3 feature ON + S3 URL not available → 503 "Not ready" + warning log

When map is ready:
- S3 feature OFF → serve from map
- S3 feature ON + S3 URL available → 301 redirect to S3
- S3 feature ON + S3 URL not available → serve from map + warning log

## Testing
Added comprehensive test suite covering all possible combinations of:
- Map ready state (ready/not ready)
- S3 redirect feature flag (ON/OFF)
- S3 URL availability (available/not available)

## Benefits
- Improved availability: Can serve tiles even when map is not ready if we have a cached version in S3
- Better resilience: Clear fallback behavior when S3 is not available
- Maintainable code: Common logic extracted into shared functions
- Full test coverage: All edge cases are tested

### CODE REVIEW

File: src/controllers/handlers.ts [hunk 1]
```diff
@@ -1,38 +1,107 @@
...
 import { isErrorWithMessage } from '../logic/error'
 import { AppComponents, Context } from '../types'
 import { ApplicationName, Feature } from '../modules/features/types'
+import { IMapComponent } from '../modules/map/types'
+
+/**
+ * Checks if the S3 redirect feature flag is enabled
+ */
+const isS3RedirectEnabled = async (features: IFeaturesComponent) => {
+  return await features.getIsFeatureEnabled(
```
Comment (LautaroPetaccio) [inline src/controllers/handlers.ts:14]:
```suggestion
  return features.getIsFeatureEnabled(
```

File: tests/controllers/handlers.spec.ts [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+          it('should serve tiles from map and log warning', async () => {
+            const handler = createLegacyTilesRequestHandler({ 
+              map: mapComponentMock, 
+              features: featuresComponentMock,
+              logs: loggerComponentMock
+            })
+            const result = await handler({ url: new URL('http://localhost') }) as HandlerResponse
+            expect(result.status).toBe(200)
+            expect(loggerWarnMock).toHaveBeenCalledWith('No S3 file available')
+            expect(result.headers?.['content-type']).toBe('application/json')
```
Comment (LautaroPetaccio) [inline tests/controllers/handlers.spec.ts:278]: This line, which is used in all tests that assert that the tiles are returned is not making sure the tiles are being returned as expected, as it only checks that the response is of type `application/json`. Would you mind changing the assertion to do so?

**Review by LautaroPetaccio — APPROVED** [review]
Looks great 👍
